### PR TITLE
[e2e-move-tests][harness] Richer assert macros

### PR DESCRIPTION
**Pull request by igor-aptos**

### CODE REVIEW

File: aptos-move/e2e-move-tests/src/harness.rs [hunk 1]
```diff
@@ -680,40 +680,92 @@ impl MoveHarness {
...
+    ($s:expr, $c:pat $(,)?) => {{
         assert!(matches!(
             $s,
             aptos_types::transaction::TransactionStatus::Keep(
                 aptos_types::transaction::ExecutionStatus::MoveAbort { code: $c, .. }
             ),
         ));
     }};
+    ($s:expr, $c:ident, $($arg:tt)+) => {{
+        assert!(
```
Comment (georgemitenkov) [inline aptos-move/e2e-move-tests/src/harness.rs:722]: nit: I think you can use `assert_matches!` directly

File: aptos-move/e2e-move-tests/src/harness.rs [hunk 2]
```diff
@@ -724,4 +776,77 @@ macro_rules! assert_move_abort {
...
+    fn test_asserts() {
+        let success = TransactionStatus::Keep(ExecutionStatus::Success);
+
+        let abort_13 = TransactionStatus::Keep(ExecutionStatus::MoveAbort {
+            code: 13,
+            location: AbortLocation::Script,
+            info: None,
+        });
+
+        assert_success!(success);
```
Comment (georgemitenkov) [inline aptos-move/e2e-move-tests/src/harness.rs:810]: nit: we can test 
```
assert_success!(
      success,
);
```
for completeness? I think this is what `$(,)?` is for

**Review by georgemitenkov — APPROVED** [review]

**Review by wrwg — APPROVED** [review]
Nice!

File: aptos-move/e2e-move-tests/src/harness.rs (hunk 1 above)
Comment (igor-aptos) [reply aptos-move/e2e-move-tests/src/harness.rs:722]: not all can (this one cannot), so leaving for consistency

**Pull request merged**